x/ibc/light-clients/07-tendermint/types/client_state.go

```diff
@@ -142,6 +142,18 @@ func (cs ClientState) ZeroCustomFields() exported.ClientState {
 	}
 }
 
+// Initialize will check that initial consensus state is a Tendermint consensus state
+// and will store ProcessedTime for initial consensus state as ctx.BlockTime()
+func (cs ClientState) Initialize(ctx sdk.Context, _ codec.BinaryMarshaler, clientStore sdk.KVStore, consState exported.ConsensusState) error {
```

Comment:
the codec is not used in any of the client states, we should remove it

Reply:
it could be used though, if the client needs to write something to its client store

Reply:
True, but this is an interface that other client-types will have to implement. I could imagine a client implementation needing a codec to correctly marshal and set metadata